plugin/src/main/java/io/snyk/eclipse/plugin/views/MenuHandler.java

```diff
@@ -3,16 +3,36 @@
 import org.eclipse.core.commands.AbstractHandler;
 import org.eclipse.core.commands.ExecutionEvent;
 import org.eclipse.core.commands.ExecutionException;
+import org.eclipse.core.resources.IProject;
+import org.eclipse.jdt.internal.core.JavaProject;
+import org.eclipse.jface.viewers.IStructuredSelection;
+import org.eclipse.ui.ISelectionService;
 import org.eclipse.ui.IWorkbenchWindow;
 import org.eclipse.ui.handlers.HandlerUtil;
 
+import io.snyk.languageserver.LsCommandID;
 import io.snyk.languageserver.protocolextension.SnykExtendedLanguageClient;
 
 public class MenuHandler extends AbstractHandler {
 
-  public Object execute(ExecutionEvent event) throws ExecutionException {
-    IWorkbenchWindow window = HandlerUtil.getActiveWorkbenchWindowChecked(event);
-    SnykExtendedLanguageClient.getInstance().triggerScan(window);
-    return null;
-  }
+	public Object execute(ExecutionEvent event) throws ExecutionException {
+		IWorkbenchWindow window = HandlerUtil.getActiveWorkbenchWindowChecked(event);
+
+		ISelectionService service = window.getSelectionService();
+		IStructuredSelection structured = (IStructuredSelection) service.getSelection();
+
+		Object firstElement = structured.getFirstElement();
+		IProject project = null;
+		
+		if (firstElement instanceof JavaProject) {
+			project = ((JavaProject) firstElement).getProject();
+		}
+
+		if (firstElement instanceof IProject) {
+			project = (IProject) firstElement;
+		}
+
+		SnykExtendedLanguageClient.getInstance().triggerScan(project);
```

Comment:
Why is this changed to a project scan? previously, we had workspaceScan and folderScan - we should not have a breaking change here.

Reply:
Confirming that both project (filepath) and workspace scans still works from the project explorer context menu.